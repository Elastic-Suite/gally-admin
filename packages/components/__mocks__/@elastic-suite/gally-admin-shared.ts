import { api } from '@elastic-suite/gally-admin-shared/src/mocks'

import catalog from '@elastic-suite/gally-admin-shared/src/mocks/static/catalog.json'
import productSortingOptions from '@elastic-suite/gally-admin-shared/src/mocks/static/product_sorting_options.json'
import docs from '@elastic-suite/gally-admin-shared/src/mocks/static/docs.json'
import entrypoint from '@elastic-suite/gally-admin-shared/src/mocks/static/index.json'
import graphql from '@elastic-suite/gally-admin-shared/src/mocks/static/graphql.json'
import metadata from '@elastic-suite/gally-admin-shared/src/mocks/static/metadata.json'
import sourceFieldOptionLabels from '@elastic-suite/gally-admin-shared/src/mocks/static/source_field_option_labels.json'
import ruleEngineGraphqlFilters from '@elastic-suite/gally-admin-shared/src/mocks/static/rule_engine_graphql_filters.json'
import ruleEngineOperators from '@elastic-suite/gally-admin-shared/src/mocks/static/rule_engine_operators.json'

export * from '@elastic-suite/gally-admin-shared/src/constants'
export * from '@elastic-suite/gally-admin-shared/src/contexts'
export * from '@elastic-suite/gally-admin-shared/src/mocks'
export * from '@elastic-suite/gally-admin-shared/src/services/breadcrumb'
export * from '@elastic-suite/gally-admin-shared/src/services/catalog'
export * from '@elastic-suite/gally-admin-shared/src/services/category'
export * from '@elastic-suite/gally-admin-shared/src/services/field'
export * from '@elastic-suite/gally-admin-shared/src/services/form'
export * from '@elastic-suite/gally-admin-shared/src/services/format'
export * from '@elastic-suite/gally-admin-shared/src/services/hydra'
export * from '@elastic-suite/gally-admin-shared/src/services/local'
export * from '@elastic-suite/gally-admin-shared/src/services/options'
export * from '@elastic-suite/gally-admin-shared/src/services/network'
export * from '@elastic-suite/gally-admin-shared/src/services/rules'
export * from '@elastic-suite/gally-admin-shared/src/services/style'
export * from '@elastic-suite/gally-admin-shared/src/services/table'
export * from '@elastic-suite/gally-admin-shared/src/services/url'
export * from '@elastic-suite/gally-admin-shared/src/services/user'
export * from '@elastic-suite/gally-admin-shared/src/types'

const body = { hello: 'world' }

/* api */
export const getApiUrl = jest.fn((url) => url)

export const fetchApi = jest.fn()
fetchApi.mockImplementation((_, resource) => {
  let data: unknown = { ...body }
  if (
    (typeof resource !== 'string' &&
      resource.title.toLowerCase() === 'metadata') ||
    (typeof resource === 'string' && resource.endsWith('metadata'))
  ) {
    data = { ...metadata }
  } else if (
    typeof resource === 'string' &&
    resource.endsWith('product_sorting_options')
  ) {
    data = { ...productSortingOptions }
  } else if (
    (typeof resource !== 'string' &&
      resource.title.toLowerCase() === 'source_field_option_label') ||
    (typeof resource === 'string' &&
      resource.endsWith('source_field_option_labels'))
  ) {
    data = { ...sourceFieldOptionLabels }
  } else if (
    (typeof resource !== 'string' &&
      resource.title.toLowerCase() === 'catalog') ||
    (typeof resource === 'string' && resource.endsWith('catalog'))
  ) {
    data = { ...catalog }
  } else if (
    typeof resource === 'string' &&
    resource.endsWith('rule_engine_graphql_filters')
  ) {
    data = { ...ruleEngineGraphqlFilters }
  } else if (
    typeof resource === 'string' &&
    resource.endsWith('rule_engine_operators')
  ) {
    data = { ...ruleEngineOperators }
  }
  return Promise.resolve(data)
})

export const removeEmptyParameters = jest.fn(
  (searchParameters = {}) => searchParameters
)

export const log = jest.fn((error, log) => log?.(error.message))

export const getApiFilters = jest.fn((x) => x)

/* bundle */

export const isVirtualCategoryEnabled = jest.fn(() => true)

/* fetch */
export const isError = jest.fn((json) => 'error' in json)
export const fetchJson = jest.fn((url) => {
  switch (url) {
    case 'http://localhost/':
      return Promise.resolve({
        json: entrypoint,
        response: {
          headers: new Headers({
            Link: '<http://localhost/docs.jsonld>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"',
          }),
        },
      })
    case 'http://localhost/docs.jsonld':
      return Promise.resolve({
        json: docs,
      })
    case 'http://localhost/graphql':
      return Promise.resolve({
        json: graphql,
      })
    default:
      return Promise.resolve({
        json: body,
        response: { headers: new Headers() },
      })
  }
})

/* form */

export const getFormValidityError = jest.fn(() => 'test')

/* graphql */
export const fetchGraphql = jest.fn(() => {
  const data: unknown = { ...body }
  return Promise.resolve(data)
})

/* parser */
export const parseSchema = jest.fn(() => Promise.resolve(api))

/* storage */
export const storageGet = jest.fn(
  () =>
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2NTk5NTM1NDQsImV4cCI6MTY1OTk4MjM0NCwicm9sZXMiOlsiUk9MRV9BRE1JTiIsIlJPTEVfQ09OVFJJQlVUT1IiXSwidXNlcm5hbWUiOiJhZG1pbkBleGFtcGxlLmNvbSJ9.ZNE2SHmtm9WgWEMmKGMjpcfEym2ysxC_CVcLtzlpuT3jlT0Zyz4VjoQHdTav9I7s9b2QzpeDNt3Sfrbka8IHcVb_TvZle_vII-TkQGal37RtXxKk_EMMQQlzLRO3zDWYy2I0ZPXv6iMlaSWOya4egOG2s1_tzFf4MnNNm2zEQbjzAtZ_Ij9UBMlSxK19AHvHlKU9AelA-NepjUdl6f4GxYPnyIwZw3PS294Wtffj7Kzw1ZZywDljl1xN_flg70cuO0GGkUqBNHrf7zBjPmXbAZZDAefoyQfVqB2v4GzDJfO847XymvBlHtCCjWx0ThK5afyOGL2MAAZkl-AoNjMcjA'
)
export const storageSet = jest.fn()
export const storageRemove = jest.fn()

/* useSchemaLoader */
export const useSchemaLoader = jest.fn(() => ({ data: 'api' }))
