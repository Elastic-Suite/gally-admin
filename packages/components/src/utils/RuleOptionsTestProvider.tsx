import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
