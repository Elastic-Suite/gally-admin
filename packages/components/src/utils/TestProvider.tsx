import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
