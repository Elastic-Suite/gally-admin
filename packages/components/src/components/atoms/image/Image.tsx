import React from 'react'
import { styled } from '@mui/system'

import { IImage, imageIconLabels } from '@elastic-suite/gally-admin-shared'

import IonIcon from '../IonIcon/IonIcon'
import { useTranslation } from 'next-i18next'

interface IProps {
  image: IImage
}

const ImageContainer = styled('div')({
  display: 'flex',
})

const ImageTag = styled('img')({
  height: 80,
  width: 80,
})

const Icons = styled('div')({
  height: '80px',
  display: 'flex',
  flexDirection: 'column',
  marginLeft: '5px',
  gap: '5px',
  flexWrap: 'wrap',
})

function Image(props: IProps): JSX.Element {
  const { image } = props
  const { t } = useTranslation('common')

  return (
    <ImageContainer>
      <ImageTag alt={t('field.productImage')} src={image.path as string} />
      <Icons>
        {image.icons
          ? image.icons.map((iconName) => (
              <IonIcon
                key={iconName}
                name={iconName}
                tooltip
                title={t(imageIconLabels[iconName])}
                style={{
                  color: '#151A47',
                  fontSize: '18px',
                  border: '1px solid #B5B9D9',
                  borderRadius: '8px',
                  padding: '3px',
                }}
              />
            ))
          : null}
      </Icons>
    </ImageContainer>
  )
}

export default Image
