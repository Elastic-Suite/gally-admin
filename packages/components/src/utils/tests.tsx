import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
