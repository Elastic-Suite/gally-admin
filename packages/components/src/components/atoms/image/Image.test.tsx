import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import Image from './Image'
import { ImageIcon } from '@elastic-suite/gally-admin-shared'

describe('Score', () => {
  it('Should match snapschot', () => {
    const { container } = renderWithProviders(
      <Image
        image={{
          path: 'media/catalog/product/v/d/vd11-ly_main.jpg',
          icons: [ImageIcon.PIN],
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
