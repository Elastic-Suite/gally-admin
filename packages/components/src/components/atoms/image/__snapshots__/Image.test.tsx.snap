// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score Should match snapschot 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <img
      alt="field.productImage"
      class="css-1qpslqa"
      src="media/catalog/product/v/d/vd11-ly_main.jpg"
    />
    <div
      class="css-133d1dw"
    >
      <ion-icon
        src="/images/ionicons/cutsom/push-pin.svg"
        style="color: rgb(13, 18, 116); width: 15px; height: 15px; display: flex; font-size: 14px; border: 1px solid #b5b9d9; border-radius: 5px; padding: 4px;"
        title="Pinned"
      />
    </div>
  </div>
</div>
`;
